tests/test_particle_sets.py

```diff
@@ -0,0 +1,156 @@
+    assert np.allclose([p.lat for p in pset], lat, rtol=1e-12)
+
+
+@pytest.mark.parametrize('mode', ['scipy', 'jit'])
+def test_pset_add_shorthand(grid, mode, npart=100):
+    lon = np.linspace(0, 1, npart, dtype=np.float32)
+    lat = np.linspace(1, 0, npart, dtype=np.float32)
+    pset = grid.ParticleSet(0, lon=[], lat=[], pclass=ptype[mode])
+    for i in range(npart):
+        pset += ptype[mode](lon=lon[i], lat=lat[i], grid=grid)
+    assert(pset.size == 100)
+    assert np.allclose([p.lon for p in pset], lon, rtol=1e-12)
+    assert np.allclose([p.lat for p in pset], lat, rtol=1e-12)
+
+
+@pytest.mark.parametrize('mode', ['scipy', 'jit'])
+def test_pset_merge_inplace(grid, mode, npart=100):
+    pset1 = grid.ParticleSet(npart, pclass=ptype[mode],
+                             lon=np.linspace(0, 1, npart, dtype=np.float32),
+                             lat=np.linspace(1, 0, npart, dtype=np.float32))
+    pset2 = grid.ParticleSet(npart, pclass=ptype[mode],
+                             lon=np.linspace(0, 1, npart, dtype=np.float32),
+                             lat=np.linspace(0, 1, npart, dtype=np.float32))
+    assert(pset1.size == 100)
+    assert(pset2.size == 100)
+    pset1.add(pset2)
+    assert(pset1.size == 200)
+
+
+@pytest.mark.xfail(reason="ParticleSet duplication has not been implemented yet")
+@pytest.mark.parametrize('mode', ['scipy', 'jit'])
+def test_pset_merge_duplicate(grid, mode, npart=100):
+    pset1 = grid.ParticleSet(npart, pclass=ptype[mode],
+                             lon=np.linspace(0, 1, npart, dtype=np.float32),
+                             lat=np.linspace(1, 0, npart, dtype=np.float32))
+    pset2 = grid.ParticleSet(npart, pclass=ptype[mode],
+                             lon=np.linspace(0, 1, npart, dtype=np.float32),
+                             lat=np.linspace(0, 1, npart, dtype=np.float32))
+    pset3 = pset1 + pset2
+    assert(pset1.size == 100)
+    assert(pset2.size == 100)
+    assert(pset3.size == 200)
+
+
+@pytest.mark.parametrize('mode', ['scipy', 'jit'])
+def test_pset_remove_index(grid, mode, npart=100):
+    lon = np.linspace(0, 1, npart, dtype=np.float32)
+    lat = np.linspace(1, 0, npart, dtype=np.float32)
+    pset = grid.ParticleSet(npart, lon=lon, lat=lat, pclass=ptype[mode])
+    for ilon, ilat in zip(lon[::-1], lat[::-1]):
+        p = pset.remove(-1)
+        assert(p.lon == ilon)
+        assert(p.lat == ilat)
+    assert(pset.size == 0)
+
+
+@pytest.mark.xfail(reason="Particle removal has not been implemented yet")
+@pytest.mark.parametrize('mode', ['scipy', 'jit'])
+def test_pset_remove_particle(grid, mode, npart=100):
+    lon = np.linspace(0, 1, npart, dtype=np.float32)
+    lat = np.linspace(1, 0, npart, dtype=np.float32)
+    pset = grid.ParticleSet(npart, lon=lon, lat=lat, pclass=ptype[mode])
+    for ilon, ilat in zip(lon[::-1], lat[::-1]):
+        p = pset.remove(pset[-1])
+        assert(p.lon == ilon)
+        assert(p.lat == ilat)
+    assert(pset.size == 0)
+
```

Comment:
I must say I don't completely understand what the difference is between `p = pset.remove(-1)` (which does work) and `p = pset.remove(pset[-1])` (which doesn't). Does the former mean we can only remove the last particle? Why would one want to use the latter indexed version rather than the former? What is the advantage?